src/DoctrineBatchUtils/BatchProcessing/SelectBatchIteratorAggregate.php

```diff
@@ -0,0 +1,138 @@
+<?php
+declare(strict_types=1);
+
+namespace DoctrineBatchUtils\BatchProcessing;
+
+use ArrayIterator;
+use DoctrineBatchUtils\BatchProcessing\Exception\MissingBatchItemException;
+use Doctrine\ORM\AbstractQuery;
+use Doctrine\ORM\EntityManagerInterface;
+use IteratorAggregate;
+use Traversable;
+
+/**
+ * 'Read' focused batching iterator that will issue a clear to the entity manager
```

Comment:
Given the description, why not call it `ReadOnlyBatchIterator`?

Reply:
I thought about calling it that you *could* reattach the entity also its very likely the entity will have setters so I thought it might be confusing

Reply:
Right, so the point is that it clears the UoW on each batch. Do we have any better naming indicating that?